zmq4.go

```diff
@@ -28,6 +30,9 @@ type Socket interface {
 	// Type returns the type of this Socket (PUB, SUB, ...)
 	Type() SocketType
 
+	// Addr returns the listen address
```

Comment:
we should probably document what happens when `Addr` is called on a socket that doesn't listen.

```suggestion
	// Addr returns the listener's address
	// Addr returns nil if the socket isn't a listener.
```